### DISCUSSION TITLE
can you share your candlestick implementation?

### DISCUSSION THREAD

**Discussion by zillionare** [h2oai/wave#1403] [Q&A]
does anybody has implemented candlestick using wave plot (or wrapped plotlyjs)? Currently I'm using plotly python package to render candlestick, then convert it to html, then load it from frame_card. The issue is performance. If I need forward/backward the bars, then a lot data need to pass between client and server. If we use wave and circular data, a lot of network transport can be saved.

**Comment by mtanco** [comment]
You might try modifying the Wave Boxplot (it's not perfect but would be effect and might suite your needs!)
<img width="897" alt="image" src="https://user-images.githubusercontent.com/4793561/166983521-6395f2de-32c0-404a-9c9b-1bc32eea51f4.png">

```python
from h2o_wave import main, app, Q, ui, data
import pandas as pd
import numpy as np


@app('/')
async def serve(q: Q):

    # Example data: https://www.kaggle.com/datasets/camnugent/sandp500ls
    df = pd.read_csv("~/Downloads/archive-2/all_stocks_5yr.csv", nrows=10)
    print(df.head())

    wave_data = df[["date", "low", "high"]]
    wave_data["box_bottom"] = np.where(df["open"] < df["close"], df["open"], df["close"])
    wave_data["box_top"] = np.where(df["open"] > df["close"], df["open"], df["close"])
    wave_data["status"] = np.where(df["open"] > df["close"], "decrease", "increase")
    print(wave_data.head())

    q.page['candlestick_plot'] = ui.plot_card(
        box='1 1 6 5',
        title='Candlestick for AAL',
        data=data(
            fields=wave_data.columns.tolist(),
            rows=wave_data.values.tolist(),
        ),
        plot=ui.plot([ui.mark(
            type='schema',
            x='=date',
            y1='=low',  # lowest value that day
            y_q1='=box_bottom',  # opening or closing price
            y_q2='=box_bottom',  # faking as candlestick plots don't have a median
            y_q3='=box_top',  # opening or closing price
            y2='=high',  # highest value that day
            color="=status",
            color_range="red green",
            color_domain=["decrease", "increase"]
        )])
    )

    await q.page.save()
```

**Comment by ddeloy** [comment]
I am a candlestick charting devotee...
Used it religiously when I was professionally trading!
Very, very cool!

**Reply by ddeloy** [reply]
Might be my entree into creating my first wave app :)

**Comment by zillionare** [comment]
@mtanco it would be great if your team can add candlestick plot. Since box plot is there, would it be easy to add candlestick?

And would it be possible to use markup card plus injected plotlyjs? However, I don't know how to pass data to markup card and finally let plotlyjs to read it.
